astrocut/make_cube.py

```diff
@@ -346,3 +348,545 @@ def make_cube(self, file_list, cube_file="img-cube.fits", sector=None, max_memor
+
+            # set up the image info table
+            cols = []
+            for kwd, val, cmt in secondary_header.cards: 
+                if type(val) == str:
+                    tpe = "S" + str(len(val))  # TODO: Maybe switch to U?
+                elif type(val) == int:
+                    tpe = np.int32
+                else:
+                    tpe = np.float64
+                    
+                # If there's already an info table, this means we are
+                # updating a cube instead of making a new one, so expanding 
+                # the length by the new FFIs (hence +self.file_list)
+
+                length = len(self.file_list)
+                """
+                if not self.update: 
+                    length = len(self.file_list)
+                else: 
+                    length = len(self.info_table)+len(self.file_list)
+                """
+                cols.append(Column(name=kwd, dtype=tpe, length=length, meta={"comment": cmt}))
+
+            cols.append(Column(name="FFI_FILE", dtype="S" + str(len(os.path.basename(self.template_file))),
+                               length=length))
+    
+            self.info_table = Table(cols)
+
+    def _build_cube_file(self, cube_file):
+        """ Build the cube file on disk with primary header, cube extension header,
+        and space for the cube, filled with zeros.
+
+        Note, this will overwrite the file if it already exists.
+
+        Parameters
+        ----------
+        cube_file : str 
+            Optional. The filename/path to save the output cube in. 
+        """
+        
+        # Making sure the output directory exists
+        direc, _ = os.path.split(cube_file)
+        if direc and not os.path.exists(direc):
+            os.makedirs(direc)
+
+        # Writing the primary header
+        hdu0 = fits.PrimaryHDU(header=self.primary_header)
+        hdul = fits.HDUList([hdu0])
+        hdul.writeto(cube_file, overwrite=True)
+
+        # Making the cube header and writing it
+        data = np.zeros((100, 100, 10, 2), dtype=np.float32)
+        hdu = fits.ImageHDU(data)
+        header = hdu.header
+        header["NAXIS4"], header["NAXIS3"], header["NAXIS2"], header["NAXIS1"] = self.cube_shape
+
+        # I think this part increases the buffer size of ext1 and writes 
+        # the cube in there?
+        with open(cube_file, 'ab') as CUBE:
+            CUBE.write(bytearray(header.tostring(), encoding="utf-8"))
+
+        # Expanding the file to fit the full data cube
+        # fits requires all blocks to be a multiple of 2880
+        cubesize_in_bytes = ((np.prod(self.cube_shape) * 4 + 2880 - 1) // 2880) * 2880
+        filelen = os.path.getsize(cube_file)
+        
+        with open(cube_file, 'r+b') as CUBE:
+            # What is this .seek?
+            # Looks like it's expanding the buffer size of the file?
+            CUBE.seek(filelen + cubesize_in_bytes - 1)
+            # b for buffer?
+            CUBE.write(b'\0')
```

Comment:
From what I understand of this, it's just seeking to the end of the `cube_file` and writing one additional null byte, `b'\0'` just being the the null character as a byte. I don't see precisely why it's being added yet, except maybe as a terminator, though from what I recall of the FITS standard only text column tables are null-terminated.

Reply:
Interesting, yeah I'm not sure why that final byte is in there, it's something I'll investigate later and add to the documentation. Thanks for clarifying!